### PR TITLE
This fixes all clippy warnings

### PR DESCRIPTION
**Pull request by dvdsk**
Made an issue for missing `is_empty()`, see:
https://github.com/RustAudio/rodio/issues/562

The DecoderImpl enum triggerd large_enum_variant. There is not an easy fix for that. We could box the VorbisDecoder but at 572 bytes I really do not think thats worth it. Annotated it with an allow

### CODE REVIEW

**Comment by dvdsk** [comment]
@est31 This is kinda a trivial PR but as I am the author, should this get a review by someone else before I merge it? 

And a followup which of the RustAudio members could ask for a review in general (not the monster that is #513 ill take care of that with a fellow engineer, but for example PR #559).

**Review by est31 — APPROVED** [review]

**Comment by dvdsk** [comment]
> Tough question. I'm generally available as a reviewer, otherwise I don't really know. Maybe we should make a call for maintainers? rodio should not die because of maintainer bottleneck :).

I have been working through the recent issues and most reporters are quite willing to work on a PR so there is a clear need for Rodio. I am in favor of looking for more maintainers, however I have no idea on how to do so safely. Rodio is a popular dependency, any mistake (or malicious intent) could have a wide impact.

**Pull request merged**
